app/assets/javascripts/admin/views/locale-switcher.js

```diff
@@ -14,24 +24,24 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
   LocaleSwitcher.prototype.setupLocaleSwitching = function () {
     var form = this.module
     var rightToLeftLocales = this.rightToLeftLocales
-    var select = form.querySelector('#attachment_locale')
-    var title = form.querySelector('.app-view-attachments__form-title input')
-    var body = form.querySelector('.app-view-attachments__form-body .app-c-govspeak-editor__textarea textarea')
-    var preview = form.querySelector('.app-view-attachments__form-body .app-c-govspeak-editor__preview')
+    var select = form.querySelector('.js-locale-switcher-selector')
+    var localeFields = form.querySelectorAll('.js-locale-switcher-field input, .js-locale-switcher-field textarea, .js-locale-switcher-custom')
 
     if (!select) {
       return
     }
 
-    select.addEventListener('change', function () {
-      if (rightToLeftLocales.indexOf(this.value) > -1) {
-        title.setAttribute('dir', 'rtl')
-        body.setAttribute('dir', 'rtl')
-        preview.setAttribute('dir', 'rtl')
+    select.addEventListener('change', function (event) {
+      var value = event.target.value
+
+      if (rightToLeftLocales.indexOf(value) > -1) {
```

Comment:
### Opinion (no action needed - feel free to ignore)

A nicer way to write this might be:

```suggestion
      if (rightToLeftLocales.includes(value)) {
```

However, `Array.prototype.includes` is not supported by IE11. I'm not sure if we've 'officially' dropped JS in IE11 yet (might be worth checking with @dnkrj). So I also understand if you'd rather leave this as is.

Reply:
Yes, I agree `includes` is more elegant but I think I'd keep the IE11 support here for the minute. From what I can see we're not using `includes` in this context anywhere else yet in Whitehall.